web/app/view/dialog/ReportConfigController.js

```diff
@@ -44,7 +44,55 @@ Ext.define('Traccar.view.dialog.ReportConfigController', {
         callingPanel.fromTime = this.lookupReference('fromTimeField').getValue();
         callingPanel.toDate = this.lookupReference('toDateField').getValue();
         callingPanel.toTime = this.lookupReference('toTimeField').getValue();
+        callingPanel.period = this.lookupReference('periodField').getValue();
         callingPanel.updateButtons();
         button.up('window').close();
+    },
+
+    onPeriodChange: function (combobox, newValue) {
+        var day, first, from = new Date(), to = new Date();
+        switch (newValue) {
+            case 'custom':
+                this.lookupReference('fromContainer').setHidden(false);
+                this.lookupReference('toContainer').setHidden(false);
+                return;
+            case 'today':
+                to.setDate(to.getDate() + 1);
+                break;
+            case 'yesterday':
+                from.setDate(to.getDate() - 1);
+                break;
+            case 'thisWeek':
+                day = from.getDay();
+                first = from.getDate() - day + (day === 0 ? -6 : 1);
+                from.setDate(first);
+                to.setDate(first + 7);
+                break;
+            case 'previousWeek':
+                day = from.getDay();
+                first = from.getDate() - day + (day === 0 ? -6 : 1);
+                from.setDate(first - 7);
+                to.setDate(first);
+                break;
+            case 'thisMonth':
+                from.setDate(1);
+                to.setDate(1);
+                to.setMonth(from.getMonth() + 1);
+                break;
+            case 'previousMonth':
+                from.setDate(1);
+                from.setMonth(from.getMonth() - 1);
+                to.setDate(1);
+                break;
+            default:
+        }
+        from.setHours(0, 0, 0, 0);
+        to.setHours(0, 0, 0, 0);
+        this.lookupReference('fromDateField').setValue(from);
+        this.lookupReference('fromTimeField').setValue(from);
+        this.lookupReference('toDateField').setValue(to);
+        this.lookupReference('toTimeField').setValue(to);
+        this.lookupReference('fromContainer').setHidden(true);
```

Comment:
I think it's better to do something like `newValue !== 'custom'` here, and remove "custom" case from switch to make it easier to read the code.